Payload String for path parameter makes impossible to build CLI tool

## Expected Behavior

We expect that the CLI tool be built without errors.

## Actual Behavior

Cannot build the CLI tool throwing this error:
```
../../gen/http/test-service/client/encode_decode.go:33:11: p.Name undefined (type string has no field or method Name)
```

## Steps to Reproduce the Problem

1. create this design at `${GOPATH}/src/foo/design/design.go`:
```go
package design

import . "goa.design/goa/http/design"
import . "goa.design/goa/http/dsl"

var _ = API("test-api", func() {
	Title("Trying to reproduce a bug")
})

var _ = Service("test-service", func() {
	HTTP(func() {
		Path("/test-service")
	})

	Method("testMethod", func() {
		Result(TestMedia)
		Payload(String)
		HTTP(func() {
			GET("/{name}")
			Response(StatusOK)
		})
	})
})

// TestPayload type: what we would expect to be POSTed
var TestPayload = Type("TestPayload", func() {
	Attribute("attribute1", String)
	Attribute("attribute2", String)
})

// TestMedia response: what should be returned when GETing
var TestMedia = ResultType("application/rdcman.test-media", func() {
	TypeName("TestMedia")
	Reference(TestPayload)

	Attributes(func() {
		Attribute("attribute1")
		Attribute("attribute2")
		Attribute("url")
	})

	View("default", func() {
		Attribute("attribute1")
		Attribute("attribute2")
	})
})
```
2. generate go code: `goa gen foo/design`
3. generate _your_ code: `goa example foo/design`
4. try to build the CLI tool: `cd cmd/testapi-cli && go build`

## Specifications

  - goa Version: 2.0.0-wip
  - Platform (Linux/Windows etc.): Linux leyla 4.17.17-100.fc27.x86_64 #1 SMP Mon Aug 20 15:53:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

--- a/http/codegen/service_data.go
+++ b/http/codegen/service_data.go
@@ -751,6 +751,7 @@ func (d ServicesData) analyze(hs *httpdesign.ServiceExpr) *ServiceData {
 			}
 			data := map[string]interface{}{
 				"PayloadRef":   payloadRef,
+				"HasFields":    design.IsObject(a.MethodExpr.Payload.Type),
 				"ServiceName":  svc.Name,
 				"EndpointName": ep.Name,
 				"Args":         args,
@@ -2402,9 +2403,9 @@ const (
 		}
 	{{- range .Args }}
 		{{- if .Pointer }}
-		if p.{{ .FieldName }} != nil {
+		if p{{ if $.HasFields }}.{{ .FieldName }}{{ end }} != nil {
 		{{- end }}
-			{{ .Name }} = {{ if .Pointer }}*{{ end }}p.{{ .FieldName }}
+			{{ .Name }} = {{ if .Pointer }}*{{ end }}p{{ if $.HasFields }}.{{ .FieldName }}{{ end }}
 		{{- if .Pointer }}
 		}
 		{{- end }}

